### PR TITLE
LuaLS Annotations + Documentation

### PR DESCRIPTION
**Pull request by Denneisk**
Annotates everything with LuaLS annotations, providing critical documentation that would otherwise require carefully tracing the execution flow of AdvDupe2.

I hope that future contributors will be able to benefit from my progressive understanding of AdvDupe2.

Let me know if this is a waste of time before I keep going, and if any changes need to be made.

### CODE REVIEW

**Comment by dvdvideo1234** [comment]
Wire team will tell you is a waste of time. I had only one patch like this ever merged :D but good effort nonetheless!

File: lua/advdupe2/sv_clipboard.lua [hunk 1]
```diff
@@ -1272,7 +1291,7 @@ local function AdvDupe2_Spawn()
 		end
 
 		AdvDupe2.SpawningEntity = true
-		local Ent = CreateEntityFromTable(v, Queue.Player)
```
Comment (thegrb93) [inline lua/advdupe2/sv_clipboard.lua:1275]: Why remove this local? Please make sure there's no code changes are in this PR if you're going to add a ton of function comments.

**Comment by thegrb93** [comment]
> Wire team will tell you is a waste of time. I had only one patch like this ever merged :D but good effort nonetheless!

These are useful comments. Your PRs had issues because you always wanted to include a bunch of cleanup along with your bugfix.

File: lua/advdupe2/sv_clipboard.lua (hunk 1 above)
Comment (Denneisk) [reply lua/advdupe2/sv_clipboard.lua:1275]: I have no clue how this happened.

**Comment by thegrb93** [comment]
@Denneisk you never marged this ready for review. Is it?

**Comment by Denneisk** [comment]
I forgot about this lol

**Pull request closed without merging**